packages/js-dash-sdk/src/SDK/Client/Platform/methods/identities/internal/createAssetLockProof.ts

```diff
@@ -15,9 +17,22 @@ export default async function createAssetLockProof(platform : Platform, assetLoc
     const account = await platform.client.getWalletAccount();
     const { dpp } = platform;
 
-    let instantLock = await account.waitForInstantLock(assetLockTransaction.hash);
     // Create poof that the transaction won't be double spend
 
+    let instantLock;
+
+    try {
+        instantLock = await account.waitForInstantLock(assetLockTransaction.hash);
+    } catch (e) {
+        // if block is mined before the transaction is instant locked instant lock won't be sent
+        if (!e.message || !e.message.startsWith('InstantLock waiting period for transaction')) {
```

Comment:
If there is an actual unhandled error happens in the DAPI, and it restarts, it will be silenced here, and I actually have no idea what would happen afterwards. Are we sure there's no other easy way but applying this hack?

Reply:
We can try adding “waitForMerkleBlock” function, listen to it alongside with the “waitForInstantLock”, and proceed with the execution in case any of them resolves.

Reply:
It won't be silence. We will have errors in DAPI log.

I'm not sure we can call it "hack" because it's normal behaviour that  instant lock could not produced by core. DIP7 describes several cases.

Reply:
👌